dataprofiler/data_readers/json_data.py

```diff
@@ -268,36 +276,36 @@ def _load_data_from_file(self, input_file_path):
                 )
             return data
 
-    def _get_data_as_records(self, data):
+    def _get_data_as_records(self, data: List) -> List[str]:
         """
         Extract the data as a record format.
 
         :param data: raw data
         :type data: list
         :return: dataframe in record format
         """
-        data = self._get_data_as_df(data)
-        data = data.to_dict(orient="records", into=OrderedDict)
-        for i, sample in enumerate(data):
-            data[i] = json.dumps(
+        data_df: pd.DataFrame = self._get_data_as_df(data)
+        data_records: List = data_df.to_dict(orient="records", into=OrderedDict)
+        for i, sample in enumerate(data_records):
+            data_records[i] = json.dumps(
                 self._convert_flat_to_nested_cols(sample), ensure_ascii=False
             )
-        return super(JSONData, self)._get_data_as_records(data)
+        return super(JSONData, self)._get_data_as_records(data_records)
 
-    def _get_data_as_json(self, data):
+    def _get_data_as_json(self, data: List) -> List[str]:
         """
         Extract the data as a json format.
 
         :param data: raw data
         :type data: list
         :return: dataframe in json format
         """
-        data = self._get_data_as_df(data)
-        data = data.to_json(orient="records")
-        char_per_line = min(len(data), self.SAMPLES_PER_LINE_DEFAULT)
-        return list(map("".join, zip(*[iter(data)] * char_per_line)))
+        data_df: pd.DataFrame = self._get_data_as_df(data)
+        data_json = data_df.to_json(orient="records")
+        char_per_line = min(len(data_json), self.SAMPLES_PER_LINE_DEFAULT)
+        return list(map("".join, zip(*[iter(data_json)] * char_per_line)))
```

Comment:
we are no longer overwriting here... was mypy complaining about something with how it was prior?

Reply:
The issue is that if we define `data` of type `A`, we can't reassign it to type `B`. If we try to change the type annotation later, mypy complains that `data` already exists. I guess a workaround would be to make `data` a union of the types it can be but I was thinking that could be messy.